docs/_includes/Types.md

```diff
@@ -1918,34 +1918,83 @@ Note that when there is no body, Dafny assumes that the *ensures*
+objects being constructed only in certain ways.  For this purpose, Dafny
 provides _constructor (method)s_, which are a restricted form of
 initialization methods.  A constructor is declared with the keyword
-`constructor` instead of `method`.  When a class contains a
-constructor, every call to `new` for that class must be accompanied
-with a call to one of the constructors.  Moreover, a constructor
+`constructor` instead of `method`.  
+When a class contains a
+constructor, every call to `new` for a class must be accompanied
+by a call to one of its constructors.  Moreover, a constructor
 cannot be called at other times, only during object creation.  Other
 than these restrictions, there is no semantic difference between using
-ordinary initialization methods and using constructors.
-
-The Dafny design allows the constructors to be named, which promotes
-using names like `InitFromList` above.  Still, many classes have just
+ordinary initialization methods and using constructors. Classes may
+declare no constructors or one or more constructors.
+
+#### Classes with no explicit constructors
+
+A class that declares no constructors has a default constructor created 
+for it. This constructor is called with the syntax
+```
+C C := new C;
+```
+This constructor simply initializes the fields of the class.
+The order of initialization is such that any field whose value depends on the
+value of another field is initialized after that other field. 
+It is an error if there are circular dependencies. 
+
+This constructor sets each class field to a default value
+of the field's type if the field declaration has no initializer 
+and to the value of the initializer expression if it does declare an initializer. 
+For the purposes of proving Dafny programs
+correct, assigning an arbitrary initial value means that the program must 
+be correct for any initial value. Compiled, executable versions of the program
+may use a specific initial value 
+(for example, but not necessarily, a zero-equivalent value).
+
+#### Classes with one or more constructors
+ 
+When one or more constructors are explicitly declared, they are named, 
+which promotes using names like `InitFromList` above.
+Constructors must have distinct names, even if their signatures are different.
+Many classes have just
 one constructor or have a typical constructor.  Therefore, Dafny
 allows one _anonymous constructor_, that is, a constructor whose name
 is essentially "".  For example:
 ```
 class Item {
+  constructor I(xy: int) // ...
   constructor (x: int, y: int)
   // ...
 }
 ```
-When invoking this constructor, the "`.`" is dropped, as in:
+The named constructor is invoked as
+```
+  Item i := new Item.I(42);
 ```
-m := new Item(45, 29);
+The anonymous constructor is invoked as
 ```
-Note that an anonymous constructor is just one way to name a
-constructor; there can be other constructors as well.
+  Item i := new Item(42, 43);
```

Comment:
This is not Dafny syntax. Drop this line.

Reply:
Fixed